fog/view/server/src/router_request_handler.rs

```diff
@@ -215,8 +202,8 @@ where
             query_responses.push(multi_view_store_query_response);
         }
 
-        shard_clients = processed_shard_response_data.shard_clients_for_retry;
-        if shard_clients.is_empty() {
+        shards = processed_shard_response_data.shards_for_retry;
+        if shards.is_empty() {
             break;
```

Comment:
Eventually we are going to want to change this termination logic to allow for the case where multiple shards are capable of serving the same range (at least, I believe that's functionality @jcape has expressed desire for). We should probably have an issue for that, as the "next step" of this PR chain.

Reply:
Ack.